cl/search/views.py

```diff
@@ -399,164 +441,167 @@ def show_results(request: HttpRequest) -> HttpResponse:
+                    next=request.path,
+                    encoded_params=quote(f"?{request.GET.urlencode()}"),
                 )
-            search_type = request.GET.get("type", SEARCH_TYPES.OPINION)
-            match search_type:
-                case SEARCH_TYPES.PARENTHETICAL:
-                    render_dict.update(do_es_search(request.GET.copy()))
-                case SEARCH_TYPES.ORAL_ARGUMENT:
-                    # Check if waffle flag is active.
-                    if waffle.flag_is_active(request, "oa-es-active"):
-                        render_dict.update(do_es_search(request.GET.copy()))
-                    else:
-                        render_dict.update(do_search(request.GET.copy()))
-                case SEARCH_TYPES.PEOPLE:
-                    if waffle.flag_is_active(request, "p-es-active"):
-                        render_dict.update(do_es_search(request.GET.copy()))
-                    else:
-                        render_dict.update(do_search(request.GET.copy()))
-                case SEARCH_TYPES.RECAP | SEARCH_TYPES.DOCKETS:
-                    if waffle.flag_is_active(request, "r-es-active"):
-                        search_results = do_es_search(request.GET.copy())
-                    else:
-                        search_results = do_search(request.GET.copy())
-                    render_dict.update(search_results)
-                case SEARCH_TYPES.OPINION:
-                    if waffle.flag_is_active(request, "o-es-active"):
-                        render_dict.update(do_es_search(request.GET.copy()))
-                    else:
-                        render_dict.update(do_search(request.GET.copy()))
-                case SEARCH_TYPES.RECAP_DOCUMENT:
-                    render_dict.update(do_es_search(request.GET.copy()))
-                case _:
-                    render_dict.update(do_search(request.GET.copy()))
-
-            # Set the value to the query as a convenience
-            alert_form.fields["name"].widget.attrs["value"] = render_dict[
-                "search_summary_str"
-            ]
-            render_dict.update({"alert_form": alert_form})
+            )
 
-            return TemplateResponse(request, "search.html", render_dict)
+        alert = get_object_or_404(
+            Alert,
+            pk=request.GET.get("edit_alert"),
+            user=request.user,
+        )
+        alert_form = CreateAlertForm(
+            instance=alert,
+            initial={"query": get_string_sans_alert},
+            user=request.user,
+        )
+    else:
+        # Just a regular search
+        if not is_bot(request):
+            async_to_sync(tally_stat)("search.results")
+
+        # Create bare-bones alert form.
+        alert_form = CreateAlertForm(
+            initial={"query": get_string, "rate": "dly"},
+            user=request.user,
+        )
+
+    search_type = request.GET.get("type", SEARCH_TYPES.OPINION)
+    search_results = {}
```

Comment:
This one also seems unused, as the match-case statement has a default case, so `search_results` will always be available.